ptf/sai_base_test.py

```diff
@@ -0,0 +1,1172 @@
+        sai_thrift_remove_bridge_port(self.client, self.port1_bp)
+        sai_thrift_remove_bridge_port(self.client, self.port0_bp)
+
+        super(SaiHelper, self).tearDown()
+
+
+class MinimalPortVlanConfig(SaiHelperBase):
+    """
+    Minimal port and vlan configuration. Create port_num bridge ports and add
+    them to VLAN with vlan_id. Configure ports as untagged
+    """
+
+    def __init__(self, port_num, vlan_id=100):
+        """
+        Args:
+            port_num (int): Number of ports to configure
+            vlan_id (int): ID of VLAN that will be created
+        """
+        super(MinimalPortVlanConfig, self).__init__()
+
+        self.port_num = port_num
+        self.vlan_id = vlan_id
+
+    def setUp(self):
+        super(MinimalPortVlanConfig, self).setUp()
+
+        if self.port_num > self.active_ports_no:
+            raise ValueError('Number of ports to configure %d is higher '
+                             'than number of active ports %d'
+                             %(self.port_num, self.active_ports_no))
+
+        self.bridge_port = []
+        self.vlan_member = []
+
+        # create bridge ports
+        for i in range(0, self.port_num):
+            bp = sai_thrift_create_bridge_port(
+                self.client, bridge_id=self.default_1q_bridge,
+                port_id=self.port_list[i], type=SAI_BRIDGE_PORT_TYPE_PORT,
+                admin_state=True)
+
+            self.assertGreater(bp, 0)
+            self.bridge_port.append(bp)
+
+        # create vlan
+        self.vlan = sai_thrift_create_vlan(self.client, vlan_id=self.vlan_id)
+        self.assertGreater(self.vlan, 0)
+
+        # add ports to vlan
+        for i in range(0, self.port_num):
+            vm = sai_thrift_create_vlan_member(
+                self.client, vlan_id=self.vlan,
+                bridge_port_id=self.bridge_port[i],
+                vlan_tagging_mode=SAI_VLAN_TAGGING_MODE_UNTAGGED)
+
+            self.assertGreater(vm, 0)
+            self.vlan_member.append(vm)
+
+        # setup untagged ports
+        for i in range(0, self.port_num):
+            status = sai_thrift_set_port_attribute(
+                self.client, self.port_list[i], port_vlan_id=self.vlan_id)
+
+            self.assertEqual(status, SAI_STATUS_SUCCESS)
+
+    def tearDown(self):
+        # revert untagged ports configuration
+        for i in range(0, self.port_num):
+            sai_thrift_set_port_attribute(
+                self.client, self.port_list[i], port_vlan_id=0)
+
+        # remove ports from vlan
+        for i in range(0, self.port_num):
+            sai_thrift_remove_vlan_member(self.client, self.vlan_member[i])
+
+        # remove vlan
+        sai_thrift_remove_vlan(self.client, self.vlan)
+
+        # remove bridge ports
+        for i in range(0, self.port_num):
+            sai_thrift_remove_bridge_port(self.client, self.bridge_port[i])
+
+        super(MinimalPortVlanConfig, self).tearDown()
+
+
+def sai_ipaddress(addr_str):
```

Comment:
You can move all the below methods to utils.py

Reply:
done